### PR TITLE
Introduce min pledge amount for stake pools

### LINKED ISSUE #903: Allow creating pools only if they own 40k+ coins

### PR DESCRIPTION
**Pull request by azarovh**
Stake pool can be created only if pledge amount is >= 40k

Closes #903

### CODE REVIEW

File: chainstate/test-suite/src/tests/pos_accounting_reorg.rs [hunk 1]
```diff
@@ -85,7 +85,7 @@ fn stake_pool_reorg(#[case] seed: Seed) {
                 .add_output(TxOutput::CreateStakePool(
                     pool_id_a,
                     Box::new(StakePoolData::new(
-                        Amount::from_atoms(rng.gen_range(100_000..200_000)),
+                        min_pledge,
```
Comment (TheQuantumPhysicist) [inline chainstate/test-suite/src/tests/pos_accounting_reorg.rs:88]: Please preserve having a range... this helps in randomizing things and detecting potential issues.

Comment (TheQuantumPhysicist) [reply chainstate/test-suite/src/tests/pos_accounting_reorg.rs:88]: This doesn't apply to this only, to be clear.

File: chainstate/test-suite/src/tests/stake_pool_tests.rs [hunk 2]
```diff
@@ -337,6 +338,47 @@ fn stake_pool_overspend(#[case] seed: Seed) {
...
+                Box::new(stake_pool_data),
+            ))
+            .build();
+        let tx_id = tx.transaction().get_id();
+
+        let result = tf.make_block_builder().add_transaction(tx).build_and_process();
+
+        assert_eq!(
+            result.unwrap_err(),
+            ChainstateError::ProcessBlockError(BlockError::StateUpdateFailed(
```
Comment (TheQuantumPhysicist) [inline chainstate/test-suite/src/tests/stake_pool_tests.rs:371]: Please add the valid case as well.

File: chainstate/tx-verifier/src/transaction_verifier/error.rs [hunk 3]
```diff
@@ -143,6 +143,10 @@ pub enum ConnectTransactionError {
     DestinationRetrievalError(#[from] SignatureDestinationGetterError),
     #[error("Output timelock error: {0}")]
     OutputTimelockError(#[from] timelock_check::OutputMaturityError),
+    #[error(
+        "Transction {0} has not enough pledge to create a stake pool: giver {1:?}, required {2:?}"
```
Comment (TheQuantumPhysicist) [inline chainstate/tx-verifier/src/transaction_verifier/error.rs:147]: Transaction

**Review by TheQuantumPhysicist — APPROVED** [review]
Looking good. Though please keep the generality of the tests. I made a comment about it. Fix it then feel free to merge. Cheers!

**Review by iljakuklic — APPROVED** [review]

**Comment by azarovh** [comment]
will merge it after the #913 PR

**Pull request merged**